jetty-server/src/test/java/org/eclipse/jetty/server/handler/FileBufferedResponseHandlerTest.java

```diff
@@ -0,0 +1,307 @@
+
+        _server = new Server();
+        HttpConfiguration config = new HttpConfiguration();
+        config.setOutputBufferSize(1024);
+        config.setOutputAggregationSize(256);
+        _local = new LocalConnector(_server, new HttpConnectionFactory(config));
+        _server.addConnector(_local);
+
+        FileBufferedResponseHandler bufferedHandler = new FileBufferedResponseHandler();
+        bufferedHandler.setTempDir(_testDir);
+        bufferedHandler.getPathIncludeExclude().include("/include/*");
+        bufferedHandler.getPathIncludeExclude().exclude("*.exclude");
+        bufferedHandler.getMimeIncludeExclude().exclude("text/excluded");
+        bufferedHandler.setHandler(_test = new TestHandler());
+
+        ContextHandler contextHandler = new ContextHandler("/ctx");
+        contextHandler.setHandler(bufferedHandler);
+
+        _server.setHandler(contextHandler);
+        _server.start();
+    }
+
+    @AfterAll
+    public static void tearDown() throws Exception
+    {
+        _server.stop();
+    }
+
+    @BeforeEach
+    public void before()
+    {
+        FS.ensureEmpty(_testDir);
+        _test._bufferSize = -1;
+        _test._mimeType = null;
+        _test._content = new byte[128];
+        Arrays.fill(_test._content, (byte)'X');
+        _test._content[_test._content.length - 1] = '\n';
+        _test._writes = 10;
+        _test._flush = false;
+        _test._close = false;
+        _test._reset = false;
+    }
+
+    public static int getNumFiles()
+    {
+        File[] files = _testDir.toFile().listFiles();
+        if (files == null)
+            return 0;
+
+        return files.length;
+    }
+
+    @Test
+    public void testPathNotIncluded() throws Exception
+    {
+        String response = _local.getResponse("GET /ctx/path HTTP/1.1\r\nHost: localhost\r\n\r\n");
+        assertThat(response, containsString(" 200 OK"));
+        assertThat(response, containsString("Write: 0"));
+        assertThat(response, containsString("Write: 7"));
+        assertThat(response, not(containsString("Content-Length: ")));
+        assertThat(response, not(containsString("Write: 8")));
+        assertThat(response, not(containsString("Write: 9")));
+        assertThat(response, not(containsString("Written: true")));
+        assertThat(response, not(containsString("NumFiles:")));
+        assertThat(getNumFiles(), is(0));
+    }
+
+    @Test
+    public void testIncluded() throws Exception
+    {
+        String response = _local.getResponse("GET /ctx/include/path HTTP/1.1\r\nHost: localhost\r\n\r\n");
+        assertThat(response, containsString(" 200 OK"));
+        assertThat(response, containsString("Write: 0"));
+        assertThat(response, containsString("Write: 9"));
+        assertThat(response, containsString("Written: true"));
+        assertThat(response, containsString("NumFiles: 1"));
+        assertThat(getNumFiles(), is(0));
+    }
+
+    @Test
+    public void testExcludedByPath() throws Exception
```

Comment:
How is this test different from `testPathNotIncluded()`?

Reply:
The `testPathNotIncluded()` is testing a path which has not been excluded but has not been included either. The `testExcludedByPath()` tests a path which is explicitly excluded.